src/poprox_recommender/handler.py

```diff
@@ -60,3 +66,49 @@ def generate_recs(event, context):
     response = {"statusCode": 200, "body": json.dumps(body)}
 
     return response
+
+def classify_news_topic(model, tokenizer, general_topics, topic):
+    inputs = tokenizer(topic, return_tensors='pt', truncation=True, padding=True, max_length=512)
+    outputs = model(**inputs)
+    logits = outputs.logits
+    probabilities = sigmoid(logits).squeeze().detach().numpy()
+    
+    # Threshold for classification.
+    threshold = 0.5
+    classified_topics = [general_topics[i] for i, prob in enumerate(probabilities) if prob > threshold]
+
+    return classified_topics
+
+def match_news_topics_to_general(articles):
+    general_topics = [
+        "US News",
+        "World News",
+        "Politics",
+        "Business",
+        "Entertainment",
+        "Sports",
+        "Health",
+        "Science",
+        "Tech ",
+        "Lifestyle",
+        "Religion",
+        "Climate",
+        "Education",
+        "Oddities",
+    ]
+    # Load the pre-trained tokenizer and model
+    tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')
+    model = BertForSequenceClassification.from_pretrained('bert-base-uncased', num_labels=len(general_topics))
+    topic_matched_dict = {} # we might be able to connect this to DB to read previous matching?
```

Comment:
We could store matched topics in the database, but if we're going to do that the code that talks to the database to retrieve and store the matches should live in `poprox-platform`. Then the mapping would get passed in as part of the request

Reply:
Yep that sounds good, shall we directly contribute to the `poprox-platform` repo to set up the logic? We'd appreciate if you can point us to the location to modify such info 😄

Reply:
That is something we may have to think about a bit, because (a) I still don't totally understand why we need matching and (b) if we do need it, then there are several places it could go. One possibility would be in the code that [extracts articles from files and loads them into the database](https://github.com/CCRI-POPROX/poprox-platform/blob/main/poprox-serverless/poprox_serverless/ingest/articles.py).

Reply:
Got it! I guess I can save the matching logic now aside while first try to extract the general topic out from all available news topic lists for Xinyi to incorporate into her changes!

Reply:
@imrecommender the new general news topic extraction function is `extract_general_topic()` in `poprox_recommender/hander.py`